### PR TITLE
Postgres render delete

### PR DESCRIPTION
**Pull request by kidonm**
fixes: https://github.com/zio/zio-sql/issues/150

This PR adds renderDelete for PostgreSQL 

- implements `DeleteExecutor`
  - makes `execute` return `Int`, the number of deleted rows
- adds `all` combinator to `DeleteBuilder` as `WHERE` clause in `DELETE FROM` is optional
- ~moves rendering functions into `Renderer` to be reusable across `render{Read | Delete | etc...}`~
- adds tests

### CODE REVIEW

File: postgres/src/main/scala/zio/sql/postgresql/PostgresModule.scala [hunk 1]
```diff
@@ -222,7 +158,92 @@ trait PostgresModule extends Jdbc { self =>
...
+  }
+
+  object PostgresFunctionDef {
+    val Sind = FunctionDef[Double, Double](FunctionName("sind"))
+  }
+
+  override def renderDelete(delete: self.Delete[_, _]): String = {
+    val renderer = Renderer()
+    renderer.buildDeleteString(delete)
+    renderer.render()
```
Comment (jczuchnowski) [inline postgres/src/main/scala/zio/sql/postgresql/PostgresModule.scala:241]: so if I did
```scala
val renderer = Renderer()
renderer.buildDeleteString(delete)
renderer.render()
renderer.buildReadString(read)
renderer.render()
```
I would get some invalid SQL in the end.
I think introducing this Renderer and two-step String rendering is overcomplicating things. Wouldn't top level `buildDeleteString` and `buildReadString` achieve essentially the same without the risk of invalid state? Or even implementing them directly in `renderDelete` and `renderRead`?
The common private functions can still be shared.

File: postgres/src/main/scala/zio/sql/postgresql/PostgresModule.scala [hunk 2]
```diff
@@ -6,14 +6,9 @@ import zio.sql.Jdbc
  */
 trait PostgresModule extends Jdbc { self =>
 
-  object PostgresFunctionDef {
-    val Sind = FunctionDef[Double, Double](FunctionName("sind"))
-  }
-
-  override def renderRead(read: self.Read[_]): String = {
-    val builder = new StringBuilder
+  sealed case class Renderer(builder: StringBuilder = new StringBuilder) {
```
Comment (jczuchnowski) [inline postgres/src/main/scala/zio/sql/postgresql/PostgresModule.scala:9]: Looks like `private final class` would be a better fit here than `sealed case class`.

File: core/jvm/src/main/scala/zio/sql/delete.scala [hunk 3]
```diff
@@ -4,6 +4,8 @@ trait DeleteModule { self: ExprModule with TableModule =>
 
   sealed case class DeleteBuilder[F[_], A, B](table: Table.Aux[A]) {
     def where[F1](expr: Expr[F1, A, Boolean]): Delete[F1, A] = Delete(table, expr)
+
+    def all[F1]: Delete[Features.Literal, A] = Delete(table, Expr.literal(true))
```
Comment (jczuchnowski) [inline core/jvm/src/main/scala/zio/sql/delete.scala:8]: So, this isn't really nice from the usability perspective. What if we removed `DeletBuilder` entirely, moved the `where` method to the `Delete` (just like `Select` has) and made `deleteFrom` return `Delete` right away (with `where` equal true)?

**Comment by jczuchnowski** [comment]
We need to figure out some details but otherwise great work here.

File: core/jvm/src/main/scala/zio/sql/delete.scala (hunk 3 above)
Comment (kidonm) [reply core/jvm/src/main/scala/zio/sql/delete.scala:8]: Let's do that.

File: postgres/src/main/scala/zio/sql/postgresql/PostgresModule.scala (hunk 1 above)
Comment (kidonm) [reply postgres/src/main/scala/zio/sql/postgresql/PostgresModule.scala:241]: Good point. Will fix it.

**Review by jczuchnowski — APPROVED** [review]

**Pull request merged**